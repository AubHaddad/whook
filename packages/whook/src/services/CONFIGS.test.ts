import initCONFIGS from './CONFIGS';
import { YError } from 'yerror';

describe('initCONFIGS', () => {
  const log = jest.fn();
  const importer = jest.fn();

  beforeEach(() => {
    log.mockReset();
    importer.mockReset();
  });

  it('should work with existing configs', async () => {
    importer.mockResolvedValueOnce({
      default: {
        CONFIG: {
          testConfig: 'test',
        },
      },
    });

    const CONFIGS = await initCONFIGS({
      NODE_ENV: 'development',
      PROJECT_SRC: '/home/whoami/my-whook-project/src',
      log,
      importer,
    });

    expect({
      CONFIGS,
      logCalls: log.mock.calls.filter((args) => 'stack' !== args[0]),
      importerCalls: importer.mock.calls,
    }).toMatchSnapshot();
  });

  it('should fail with non-existing file', async () => {
    importer.mockImplementationOnce(() => {
      throw new Error('EEXISTS');
    });

    try {
      await initCONFIGS({
        NODE_ENV: 'development',
        PROJECT_SRC: '/home/whoami/my-whook-project/src',
        log,
        importer,
      });
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: (err as YError).code,
        errorParams: (err as YError).params,
        logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
        importerCalls: importer.mock.calls,
      }).toMatchSnapshot();
    }
  });
});
