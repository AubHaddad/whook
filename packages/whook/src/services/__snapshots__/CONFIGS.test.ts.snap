// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initCONFIGS should fail with non-existing file 1`] = `
Object {
  "errorCode": "E_NO_CONFIG",
  "errorParams": Array [
    "/home/whoami/my-whook-project/src/config/development/config.js",
  ],
  "importerCalls": Array [
    Array [
      "/home/whoami/my-whook-project/src/config/development/config.js",
    ],
  ],
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the CONFIGS service.",
    ],
    Array [
      "warning",
      "‚ö° - Loading configurations from \\"/home/whoami/my-whook-project/src/config/development/config.js\\".",
    ],
    Array [
      "warning",
      "‚ò¢ - Could not load configuration file \\"/home/whoami/my-whook-project/src/config/development/config.js\\".",
    ],
  ],
}
`;

exports[`initCONFIGS should work with existing configs 1`] = `
Object {
  "CONFIGS": Object {
    "CONFIG": Object {
      "testConfig": "test",
    },
  },
  "importerCalls": Array [
    Array [
      "/home/whoami/my-whook-project/src/config/development/config.js",
    ],
  ],
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the CONFIGS service.",
    ],
    Array [
      "warning",
      "‚ö° - Loading configurations from \\"/home/whoami/my-whook-project/src/config/development/config.js\\".",
    ],
  ],
}
`;
